Minor corrections in docs for reserved memory pool (#1226)
prestosql/presto: presto-docs/src/main/sphinx/admin/spill.rst

When user memory pool is exhausted then a single query will be promoted to a reserved pool.
In such case only that query is allowed to progress thus reducing cluster
concurrency. Revocable memory will try to prevent that by triggering spill.
Reserved pool is of ``query.max-total-memory-per-node`` size. This means that
when ``query.max-total-memory-per-node`` is large then user memory pool might be
much smaller than ``query_max_memory_per_node``. This will cause excessive
spilling for queries that consume large amounts of memory per node.
Such queries could finish much quicker when spill is disabled because they

> execute in reserved pool but it could reduce the cluster concurrency drastically.

Reviewer: ```suggestion
execute in reserved pool. However, this could also significantly reduce cluster concurrency.
```
